src/psyclone/psyir/nodes/loop.py

```diff
@@ -362,19 +391,31 @@ def kernel(self, kern):
         self._kern = kern
 
     @property
-    def variable_name(self):
+    def variable(self):
         '''
-        :returns: the name of the control variable for this loop.
-        :rtype: str
+        :returns: a reference to the control variable for this loop.
+        :rtype: :py:class:`psyclone.psyir.symbols.DataSymbol`
+        '''
```

Comment:
I like that the create and setter methods don't accept None, but since the init still does this property can propagate a None value or usually fail with an error like name is not a valid method of NoneType(as it is often used as `loop.variable.name`). I think we need to check if it is None here and raise an error saying the Loop is still incomplete or something similar.

Reply:
Yes that's a great idea. As agreed I've added the `_check_variable` method and updated the tests. The only downside I can see is that testing whether the variable has been set will require exception handling, but we don't do that anywhere at the moment.